blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/ArtifactImpl.java

```diff
@@ -0,0 +1,44 @@
+package io.jenkins.blueocean.service.embedded.rest;
+
+import hudson.model.Run;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BlueArtifact;
+import org.kohsuke.stapler.Stapler;
+
+public class ArtifactImpl extends BlueArtifact {
+    final private Run run;
+    final private Run.Artifact artifact;
+    public ArtifactImpl(Run run, Run.Artifact artifact) {
```

Comment:
Pass 'Reachable parent' parameter so that you can compute it's reference like you do in ArtifactContainerImpl.java.